InstallWorkload: Use the repo's `nuget.config` as the base for installing manifests

radical opened the pull request:
Install workloads task: Use root nuget.config to construct the  …
.. nuget.config used for restoring the workload manifests.

An alternative fix to dotnet@1091ed9

lewing reviewed (approved):
looks good, should we open an issue about sharing the target?

radical committed:
Install workloads task: Use root nuget.config to construct the

.. nuget.config used for restoring the workload manifests.

An alternative fix to https://github.com/dotnet/runtime/pull/60495/commits/1091ed9f57ca814b4046a860d36d3acf94bf4def
